project/driver_controls/src/pedal_fsm.c

```diff
@@ -0,0 +1,56 @@
+#include "pedal_fsm.h"
+#include "input_event.h"
+
+// Pedal FSM state definitions
+
+FSM_DECLARE_STATE(state_brake);
+FSM_DECLARE_STATE(state_coast);
+FSM_DECLARE_STATE(state_driving);
+FSM_DECLARE_STATE(state_cruise_control);
+
+// Pedal FSM transition table definitions
+
+FSM_STATE_TRANSITION(state_brake) {
+  FSM_ADD_TRANSITION(INPUT_EVENT_GAS_COAST, state_coast);
+  FSM_ADD_TRANSITION(INPUT_EVENT_GAS_PRESSED, state_driving);
+}
+
+FSM_STATE_TRANSITION(state_coast) {
+  FSM_ADD_TRANSITION(INPUT_EVENT_GAS_PRESSED, state_driving);
+  FSM_ADD_TRANSITION(INPUT_EVENT_GAS_BRAKE, state_brake);
+  FSM_ADD_TRANSITION(INPUT_EVENT_CRUISE_CONTROL, state_cruise_control);
+}
+
+FSM_STATE_TRANSITION(state_driving) {
+  FSM_ADD_TRANSITION(INPUT_EVENT_GAS_BRAKE, state_brake);
+  FSM_ADD_TRANSITION(INPUT_EVENT_GAS_COAST, state_coast);
+  FSM_ADD_TRANSITION(INPUT_EVENT_CRUISE_CONTROL, state_cruise_control);
+}
+
+FSM_STATE_TRANSITION(state_cruise_control) {
+  FSM_ADD_TRANSITION(INPUT_EVENT_GAS_BRAKE, state_brake);
+  FSM_ADD_TRANSITION(INPUT_EVENT_CRUISE_CONTROL, state_brake);
+}
+
+// Pedal FSM arbiter function
+
+static bool prv_check_pedal(const Event *e) {
+  return true;
+}
+
+// Pedal FSM output functions
+
+static void prv_state_output(FSM *fsm, const Event *e, void *context) {
```

Comment:
How is the pedal state actually processed? Where would we actually process the data from the pedal and send our CAN message?